openquake/hazardlib/sourceconverter.py

```diff
@@ -654,10 +654,21 @@ def convert_mfdist(self, node):
                     min_mag=mfd_node['minMag'], bin_width=mfd_node['binWidth'],
                     occurrence_rates=~mfd_node.occurRates)
             elif mfd_node.tag.endswith('truncGutenbergRichterMFD'):
-                return mfd.TruncatedGRMFD(
-                    a_val=mfd_node['aValue'], b_val=mfd_node['bValue'],
-                    min_mag=mfd_node['minMag'], max_mag=mfd_node['maxMag'],
-                    bin_width=self.width_of_mfd_bin)
+                slip_rate = mfd_node.get('slipRate')
+                rigidity = mfd_node.get('rigidity')
+                if slip_rate:
+                    assert rigidity
+                    # instantiate with a NaN area, to be fixed later on
```

Comment:
Why not simply set NaN the MFD?

Reply:
I need to keep track of the parameters .slip_rate and .rigidity to be used a few lines below